src/Handler/StreamHandler.php

```diff
@@ -308,9 +308,17 @@ function () use ($context, $params) {
         );
 
         return $this->createResource(
-            function () use ($request, &$http_response_header, $context) {
+            function () use ($request, &$http_response_header, $context, $options) {
                 $resource = fopen((string) $request->getUri()->withFragment(''), 'r', null, $context);
                 $this->lastHeaders = $http_response_header;
+
+                if (isset($options['read_timeout'])) {
+                    $readTimeout = $options['read_timeout'];
+                    $sec = (int) $readTimeout;
+                    $usec = ($readTimeout - $sec) * 1000000;
```

Comment:
Wouldn't this always be 0? If a user supplies a readTimeout of `1`, then `$usec` will be set to `(1 - 1) * 1000000`.